system/lib/sbrk.c

```diff
@@ -29,25 +29,17 @@
 #define SET_ERRNO()
 #endif
 
-// FIXME just use  = &sbrk_val  here. That is simpler, but currently it has a
-// code size cost for tiny programs that don't use malloc. We link in malloc by
-// default, and depend on metadce to remove it when not used, which we can do,
-// but a static assignment here would mean a nonzero value in a memory segment,
-// which metadce cannot remove (it can remove code, but not data in segments).
-// The current code allows tiny programs without malloc to have no data segments
-// at all, while programs that use malloc may suffer a few more bytes of code
-// size, but that's negligible compared to malloc itself.
-// TODO: when we stop including malloc by default we can use the simpler way.
-static intptr_t sbrk_val = 0;
+extern size_t __heap_base;
+
+static intptr_t sbrk_val = (intptr_t)&__heap_base;
 
 intptr_t* emscripten_get_sbrk_ptr() {
-  extern size_t __heap_base;
-  intptr_t* sbrk_ptr = &sbrk_val;
-#if __EMSCRIPTEN_PTHREADS__
-  if (__c11_atomic_load((_Atomic(intptr_t)*)sbrk_ptr, __ATOMIC_SEQ_CST) == 0) {
-    __c11_atomic_store((_Atomic(intptr_t)*)sbrk_ptr, (intptr_t)&__heap_base, __ATOMIC_SEQ_CST);
-  }
-#else
+#ifdef __USING_EMSCRIPTEN_RELOCATABLE__
```

Comment:
You can just use `__PIC__` here instead I think.

Reply:
Thanks! I was wondering if there was something, but couldn't find a `__has_feature()` option for it, but I guess I was looking in the wrong place...